### PR TITLE
NO-SNOW: various fixes/improvements in the SDK

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**
This PR contains multiple fixes/improvements:
- Include client version in the failure event
- Add some basic verifications in the generated EP
- Add logging to debug a stats mismatch issue
- Add server side failure status code to client telemetry as well

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/streaming/internal/EpInfo.java [hunk 1]
```diff
@@ -17,6 +21,30 @@ class EpInfo {
     this.columnEps = columnEps;
   }
 
+  /** Some basic verification logic to make sure the EP info is correct */
+  public void verifyEpInfo() {
```
Comment (sfc-gh-tjones) [inline src/main/java/net/snowflake/ingest/streaming/internal/EpInfo.java:25]: FYI we have this check server side now as well. We could always remove it here if you want?

Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/internal/EpInfo.java:25]: This is before the file is built and customer can see it right away so they can save the log and report it, it doesn't hurt to verify it twice but i feel like adding the check at server side is too late.

Comment (sfc-gh-lsembera) [inline src/main/java/net/snowflake/ingest/streaming/internal/EpInfo.java:25]: We could have more here: min is always smaller than max (both int and string), neither min nor max strings are longer than 32 bytes (in hex),

**Review by sfc-gh-lsembera — APPROVED** [review]

File: src/main/java/net/snowflake/ingest/streaming/internal/EpInfo.java (hunk 1 above)
Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/internal/EpInfo.java:25]: We could add these later as we go, will merge this first to have a new release

**Pull request merged**